src/Swift.Bindings/src/Emitter/StringEmitter/Handler/TypeHandler.cs

```diff
@@ -74,19 +74,18 @@ public void Emit(CSharpWriter csWriter, SwiftWriter swiftWriter, IEnvironment en
             csWriter.WriteLine($"public unsafe struct {structDecl.Name} : {typeof(ISwiftObject).Name} {{");
             csWriter.Indent++;
 
-            // Emit each field in the struct
-            foreach (var fieldDecl in structDecl.Fields)
+            foreach (PropertyDecl propertyDecl in structDecl.Properties)
             {
-                string accessModifier = fieldDecl.Visibility == Visibility.Public ? "public" : "private";
-                var fieldRecord = env.TypeDatabase.GetTypeRecordOrThrow(fieldDecl.SwiftTypeSpec);
-                csWriter.WriteLine($"{accessModifier} {fieldRecord.CSTypeIdentifier} {fieldDecl.Name};");
-
-                // TODO: Fix memory access violation
-                // // Verify field against Swift type information
-                // if (swiftTypeInfo.HasValue && !VerifyFieldRecord(swiftTypeInfo.Value, structDecl.Fields.IndexOf(fieldDecl), fieldDecl))
-                // {
-                //     Console.WriteLine("Field record does not match the field declaration");
-                // }
+                if (conductor.TryGetPropertyHandler(propertyDecl, out var propertyHandler))
+                {
+                    var fieldRecord = env.TypeDatabase.GetTypeRecordOrThrow(propertyDecl.SwiftTypeSpec);
+                    csWriter.WriteLine($"private {fieldRecord.CSTypeIdentifier} {propertyDecl.Name}_;");
```

Comment:
Can you add an explanation to why we need to emit the field to C#?
I guess it's because of the type layout for frozen structs, but it would be good to document that. Also such fields are never assigned and never accessed... which is another interesting quirck which should be documented - maybe even in the generated code (accessing the field directly will not do the right thing).

Aside - this means that I can't have a property without a backing field in swift?